src/Terrain/terrain.cpp

```diff
@@ -473,25 +473,25 @@ float Terrain::get_H_cost(std::array<float, 3> xyz1, std::array<float, 3> xyz2)
     double D1 = 1.0;
     double D2 = 1.414;
     double D3 = 1.0;
-    auto [x1, y1, z1] = xyz1;
-    auto [x2, y2, z2] = xyz2;
+    //auto [x1, y1, z1] = xyz1;
+    //auto [x2, y2, z2] = xyz2;
 
-    float DX = abs(x1 - x2);
-    float DY = abs(y1 - y2);
-    float DZ = abs(z1 - z2);
+    float DX = abs(xyz1[0] - xyz2[0]);
+    float DY = abs(xyz1[1] - xyz2[1]);
+    float DZ = abs(xyz1[2] - xyz2[2]);
 
     return (DZ * D3 + abs(DX - DY) * D1 + D2 * std::min(DX, DY));
 }
 float Terrain::get_H_cost(std::array<int, 3> xyz1, std::array<int, 3> xyz2) {
     double D1 = 1.0;
     double D2 = 1.414;
     double D3 = 1.0;
-    auto [x1, y1, z1] = xyz1;
-    auto [x2, y2, z2] = xyz2;
+    //auto [x1, y1, z1] = xyz1;
+    //auto [x2, y2, z2] = xyz2;
```

Comment:
What's wrong with destructuring? Did you set your C++ version?

https://en.cppreference.com/w/cpp/language/structured_binding